[RFC] [sw/silicon_creator] External flash format specification

sameo opened the pull request:
Specify the external flash format for OpenTitan integrated use cases.

This document describes the partition, partition table, bundle, key and asset manifests structure for supporting integrated use cases where:

- The external flash is not trusted
- The external flash may hold multiple, non OpenTitan pieces of firmware data.

> ***Note*** The `rom_ext` documentation may not be the right place for specifying this flash format, but this is where the current Earlgrey format is defined.

mvaquez committed:
[sw/silicon_creator] External flash format specification

Specify the external flash format for OpenTitan integrated use cases.

This document describes the partition, partition table, bundle, key and
asset manifests structure for supporting integrated use cases where:

- The external flash is not trusted
- The external flash may hold multiple, non OpenTitan pieces of firmware
  data.

Signed-off-by: Mathieu Vaquez <mathieu@rivosinc.com>
Signed-off-by: Samuel Ortiz <sameo@rivosinc.com>
Co-authored-by: Emmanuel Puerto <emmanuel@rivosinc.com>
Co-authored-by: Loic Lefort <loic@rivosinc.com>

msfschaffner commented:
@sameo I think this document should be labelled as an RFC. Can you rename the PR title accordingly?
We can put it up for discussion in the next working group meeting if that works for you...

CC @moidx

sameo commented:
> @sameo I think this document should be labelled as an RFC. Can you rename the PR title accordingly? 

I agree. Should have put it there in the first place.

> We can put it up for discussion in the next working group meeting if that works for you...

Yes, that works for me.

loiclefort committed:
[sw/silicon_creator] External flash format: update firmware_desc_t

- add virtual_addr to firmware_desc_t
- clarify the actual firmware size and which part of the asset is loaded.

Signed-off-by: Loïc Lefort <loic@rivosinc.com>

loiclefort committed:
[sw/silicon_creator] External flash format: switch to ECDSA for signatures

Signed-off-by: Loïc Lefort <loic@rivosinc.com>

rswarbrick commented:
Since this PR has not been reviewed by anyone in the project and it is documenting an RFC that I don't believe has been adopted, I'm closing it. Please re-open if I've misunderstood.
